### ISSUE TITLE
`FunctionBase::getInlines()` Assertion fails for Fortran binary build

### ISSUE THREAD

**Issue by Prapti-044** [dyninst/dyninst#1472]
**Intention**
I want to get inline getter that works for both gcc and fortran.

**Describe the bug**

I built [codesign-kernels](https://github.com/E3SM-Project/codesign-kernels/tree/master/nested_loops) with the following command: `make nested "FC=gfortran" "FFLAGS= -O3" "LDFLAGS= -O3"`. Then I ran dyninst on the binary to get information about the disassembly for [my project](https://github.com/Prapti-044/dis-viz). It works perfectly for any C project, but for this fortran project, it gets error while looking for the inlines from the binary. The minimal example that works for any C++ build, but fails for [my sample fortran](https://github.com/E3SM-Project/codesign-kernels/tree/master/nested_loops) at the call of `topLevelFunc->getInlines()`. 

```c++
#include <CodeObject.h>
#include <Function.h>
#include <InstructionDecoder.h>
#include <Symtab.h>

namespace InstructionAPI = Dyninst::InstructionAPI;
namespace ParseAPI = Dyninst::ParseAPI;
namespace SymtabAPI = Dyninst::SymtabAPI;

int main() {
  std::string binaryPath =
      "/home/prapti/RA_Kate/codesign-kernels/nested_loops/nested";
  SymtabAPI::Symtab *symtab;
  auto isParsable = SymtabAPI::Symtab::openFile(symtab, binaryPath);
  if (!isParsable) {
    std::cerr << "Error: file " << binaryPath << " can not be parsed"
              << std::endl;
    return -1;
  }
  auto sts = std::make_unique<ParseAPI::SymtabCodeSource>(
      const_cast<char *>(binaryPath.c_str()));
  auto co = std::make_unique<ParseAPI::CodeObject>(sts.get());
  co->parse();
  auto funcs = co->funcs();

  // Inlines
  for (const auto f : funcs) {
    auto topLevelFuncs = std::set<SymtabAPI::FunctionBase *>();
    for (const auto &block : f->blocks()) {
      SymtabAPI::Function *symt_func = nullptr;
      symtab->getContainingFunction(block->start(), symt_func);
      if (!symt_func)
        continue;
      topLevelFuncs.insert(symt_func);
    }
    auto inlineFuncs = std::set<SymtabAPI::InlinedFunction *>();
	for (auto &topLevelFunc : topLevelFuncs) {
		auto ic = topLevelFunc->getInlines(); // The error is given here
	}
  }
}
```

The error I am getting is the following:

```plain
 /usr/include/boost/smart_ptr/shared_ptr.hpp:783: typename boost::detail::sp_member_access<T>::type boost::shared_ptr<T>::operator->() const [with T = Dyninst::SymtabAPI::typeSubrange; typename boost::detail::sp_member_access<T>::type = Dyninst::SymtabAPI::typeSubrange*]: Assertion `px != 0' failed.
fish: Job 1, './DisViz -b /home/…' terminated by signal SIGABRT (Abort)
```

**Expected behavior**
I know that ftn probably does not optimize for inline functions, but c compilers do. What I expect is that it will throw an exception if it can't get the inlines from the `BaseFunction`. So that I can catch and ignore it. 

**System:**
- Platform: x86_64
- Commit: 8fa753d955d43747aa46b0a528a81a0d9adf677d
- ftn: gfortran
- GCC: 13

**Comment by hainest** [comment]
Inline information is only available when debugging information is present. From the codesign [Makefile](https://github.com/E3SM-Project/codesign-kernels/blob/master/nested_loops/Makefile), it looks like you might have to manually pass "-g". Could you rebuild `nested` using `"FFLAGS= -O3 -g"`?

**Comment by Prapti-044** [comment]
@hainest , I have also tried with `make nested "FC=gfortran" "FFLAGS= -O3 -g" "LDFLAGS= -O3 -g"`, but same error.

When compiling a C++ file without `-g`, dyninst actually runs properly without error (it just does not show enough information about the binary). In any case, should it really give signal SEGABRT and crash?

**Comment by hainest** [comment]
No, we don't want to crash in any situation. I just wanted to check if it was a lack of DWARF that was causing the problem. I'll see if I can reproduce locally. Just to make sure we're seeing the same issue, could you post a gdb backtrace from the application where you saw the crash?
